consensus/bitcoin-ng.md

```diff
@@ -0,0 +1,87 @@
+
+Latency
+==
+
+On average in Bitcoin blocks are broadcasted every 10 minutes - it is driven by protocol and guarded by target. Aeternity assumed 3 minutes for Key Blocks and 3 seconds time delta between Micro Blocks. 3 minutes resolution of Key Blocks is also guarded by target and Proof-of-Work.
+
+Rewards
+==
+There is major challenge in providing liveness and integrity of the chain in the scheme proposed by Bitcoin-NG. It is solved by crypto-economic incentives. To guarantee that the next leader follows as much as possible of Micro Blocks the next leader receives fixed award on the top of 60% of fees from the previous generation of Micro Blocks. The previous leader receives 40% of the fees included in transactions he confirmed.
+
+     ┌────────┐                        ┌────────┐
+     │        │   ┌─────┐   ┌─────┐    │        │  ┌─────┐
+     │Leader 1│───│FeeN1│───│FeeN2│────│Leader 2├──│FeeN3│
+     │        │   └─────┘   └─────┘    │        │  └─────┘
+     └────────┘                        └────────┘
+      Leader 2 Award: Governance(Coinbase) + 0.6*(FeeN1+FeeN2) + 0.4*FeeN3
+
+
+Forks
+==
+
+Bitcoin-NG inherits Bitcoin's ability to fork in case of multiple block producers at the same height. It happens on Key Block level. Bitcoin-NG resolves the conflict using the same method as Bitcoin - that is - the longest chain survives. Micro Blocks carry no weight. It is still possible to have forks at Micro Block level.
+1. Network propagation time. There is a delay between mining new Key Block and receiving it by the old leader. In that time the old leader still broadcasts micro blocks. They are not valued any more, though. New leader will re-sign them and re-broadcast. We expect such Micro forks at every Leader switch.
+
+2. Malicious leader. It is possible that malicious actor will emit conflicting Micro Blocks. The result will be incorrect chain. To fix that we will emit Proof-of-Fraud.
+
+
+                         ┌──────┐
+                ┌────────│MB(1)'│
+                │        └──────┘
+                │
+                │
+                │       ┌────────┐
+             ┌─────┐    │        │  ┌─────┐
+          ───│MB(1)│────│  KB-2  ├──│MB(2)│
+             └─────┘    │        │  └─────┘
+                        └────────┘
+
+        Micro Block broadcasted by old leader is no longer forming consensus.
+        New leader may read it and include all transaction in newly broadcasted
+        micro blocks. Most likely, the new leader will broadcast micro blocks
+        as soon as possible. Given that nodes in the network have very similar
+        mempool content MB(1)' and MB(2) should be very similar. Re-signing
+        the delta of transactions missing in Leader 2's node is very cheap operation.
```

Comment:
How cheap is it really? We have a new height, etc. so there is validation necessary?!